droidlet/interpreter/robot/tests/fake_agent.py

```diff
@@ -445,13 +443,8 @@ def init_controller(self):
     def perceive(self, force=False):
         super().perceive(force=force)
         self.perception_modules["self"].perceive(force=force)
-        new_state = self.perception_modules["vision"].perceive(force=force)
-        if new_state is not None:
-            new_objects, updated_objects = new_state
-            for obj in new_objects:
-                obj.save_to_memory(self.memory)
-            for obj in updated_objects:
-                obj.save_to_memory(self.memory, update=True)
+        perception_output = self.perception_modules["vision"].perceive(force=force)
```

Comment:
the previous code let you understand what is returned, and you can further modify things here incrementally if you wanted.

the new opaque dict doesn't give that flexibility. I think we should return separate outputs (like `new_objects, updated_objects, humans = ...perceive(...)` )

Reply:
Fixed in the latest commit